webpack/components/AnsibleRolesSwitcher/AnsibleRolesSwitcherSelectors.js

```diff
@@ -0,0 +1,62 @@
+import { differenceBy, slice, includes, uniq } from 'lodash';
+import Immutable from 'seamless-immutable';
+import { createSelector } from 'reselect';
+
+const compare = (a, b) => {
+  if (a.name < b.name) {
+    return -1;
+  }
+  if (a.name > b.name) {
+    return 1;
+  }
+  return 0;
+};
+
+const switcherState = state => state.foremanAnsible.ansibleRolesSwitcher;
+
+const markInheritedRoles = (roles, inheritedRoleIds) => roles.map(role => (
+  includes(inheritedRoleIds, role.id) ?
+    { ...role, inherited: true } :
+    role
+));
+
+export const selectResults = state =>
+  Immutable.asMutable(uniq(switcherState(state).results)).sort(compare);
```

Comment:
Why does it need to be `mutable`? If you are going to break to immutability, IMO it is better to do it in the component.

Reply:
because `sort` mutates original array. I wrap it with `Immutable` to keep the immutability.